scripts/jest/matchers/toWarnDev.ts

```diff
@@ -12,12 +12,12 @@ declare global {
   namespace jest {
     // eslint-disable-next-line @typescript-eslint/generic-type-naming
     interface Matchers<R> {
-      toWarnDev(expectedMessage?: string): R;
+      toWarnDev(expectedMessage?: string): CustomMatcherResult;
     }
   }
 }
 
-const matcher = {
+const matcher: jest.ExpectExtendMap = {
```

Comment:
this file change seems unrelated, but would be `chore` otherwise, so fine likely

Reply:
I converted `setupTest.js` to TypeScript and it raised some TS issues so I took this opportunity to fix them.